src/pages/assistant/index.tsx

```diff
@@ -47,50 +41,46 @@ export default AssistantPage;
 export const getServerSideProps = withGlobalData(
   withAssistantClient<AssistantLayoutProps & AssistantContentProps>(
     async function ({ client, query }) {
-      const transportModeFilter = parseFilterQuery(query.filter);
-
-      let initialFromFeature;
-      if (query.fromLat && query.fromLon && query.fromLayer) {
-        const fromLat = parseFloat(query.fromLat.toString());
-        const fromLon = parseFloat(query.fromLon.toString());
-        const fromLayer = parseLayerQueryString(query.fromLayer.toString());
-
-        initialFromFeature = await client.reverse(fromLat, fromLon, fromLayer);
-      }
-
-      let initialToFeature;
-      if (query.toLat && query.toLon && query.toLayer) {
-        initialToFeature = await client.reverse(
-          parseFloat(query.toLat.toString()),
-          parseFloat(query.toLon.toString()),
-          parseLayerQueryString(query.toLayer.toString()),
+      const tripQuery = parseTripQuery(query);
+      if (tripQuery) {
+        const from = await client.reverse(
+          parseFloat(tripQuery.fromLat),
```

Comment:
What happens here if `fromLat`, `fromLon` etc is not parsable? Meaning if someone changes the URL. And why should we have to parse to float right after parsing the entire query (in `parseTripQuery(query)`)? I would expect after executing `parseTripQuery(query)` everything I needed was safely sanitised and ready to use.